src/Agent.Worker/ExecutionContext.cs

```diff
@@ -19,13 +19,14 @@ public interface IExecutionContext : IAgentService
         TaskResult? Result { get; set; }
         TaskResult? CommandResult { get; set; }
         CancellationToken CancellationToken { get; }
+        Int32 TimeoutInMinutes { get; }
```

Comment:
I am not sure whether we need add the Timeout into the ExecutionContext. 
it's a step level control concept, like enable, alwaysrun and continueonerror.
I think you should handle timeout within step runner instead of execution context.